src/main/java/bio/terra/service/dataset/flight/unlock/DatasetUnlockFlight.java

```diff
@@ -0,0 +1,46 @@
+package bio.terra.service.dataset.flight.unlock;
+
+import static bio.terra.common.FlightUtils.getDefaultRandomBackoffRetryRule;
+
+import bio.terra.app.configuration.ApplicationConfiguration;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.auth.iam.IamResourceType;
+import bio.terra.service.common.JournalRecordUpdateEntryStep;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.flight.UnlockDatasetStep;
+import bio.terra.service.dataset.flight.lock.DatasetLockSetResponseStep;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.journal.JournalService;
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.RetryRule;
+import java.util.UUID;
+import org.springframework.context.ApplicationContext;
+
+public class DatasetUnlockFlight extends Flight {
+  public DatasetUnlockFlight(FlightMap inputParameters, Object applicationContext) {
+    super(inputParameters, applicationContext);
+    ApplicationContext appContext = (ApplicationContext) applicationContext;
+    ApplicationConfiguration appConfig = appContext.getBean(ApplicationConfiguration.class);
+    DatasetService datasetService = appContext.getBean(DatasetService.class);
+    JournalService journalService = appContext.getBean(JournalService.class);
+
+    // Input parameters
+    UUID datasetId =
+        UUID.fromString(inputParameters.get(JobMapKeys.DATASET_ID.getKeyName(), String.class));
+    AuthenticatedUserRequest userReq =
+        inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    String lockName = inputParameters.get(JobMapKeys.REQUEST.getKeyName(), String.class);
+
+    // Configurations
+    RetryRule unlockDatasetRetry =
+        getDefaultRandomBackoffRetryRule(appConfig.getMaxStairwayThreads());
+
+    // Steps
+    addStep(new UnlockDatasetStep(datasetService, datasetId, false, lockName), unlockDatasetRetry);
```

Comment:
I'm sorry, I vaguely remember that we talked about this, but I can't remember what we decided on!

The way this currently works, an admin, steward, or custodian could easily remove a lock that is rightfully present for an actively running job.

One possible additional layer of safety is to expand the corresponding unlock endpoint to take in an optional boolean query parameter `shouldFailOnRunningJob` (or a better name) that defaults to true.  Then we could conditionally add in another step that fails if the lockName corresponds to a running job before unlocking.

This would allow a user to remove a lock held by a running job intentionally, but make it harder to do so accidentally.

Thoughts?

Reply:
I don't remember talking about this, but this is definitely a good point. 

I like the idea of the flag because then we could hand the user the flight status. 
I do think this is a bit more involved - I was thinking we would first check to make sure the dataset is actually exclusively locked by the flight in question. Then, we would grab the flight status. Normally, we do auth checks in the job service, but I think that's unnecessary since we have already checked that the user has permission on the dataset in question and confirmed that the job is associated with this dataset. 
At any rate, I've gotten started on this, but this is going to take a little more time. I'll pick this back up next week. 

Alternatively, we could just add a warning message on the endpoint and assume that only power users will be using this endpoint.

Reply:
Thanks so much for working on this, and sorry for throwing in more complexity!

> Alternatively, we could just add a warning message on the endpoint and assume that only power users will be using this endpoint.

I would be uneasy with relying on this if we expose this functionality in the TDR UI, which I suspect we're likely to do at some point.

Reply:
Ok, this idea has been implemented in the form of the "forceUnlock" flag on unlocking a dataset or snapshot.